reana_server/validation.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA Server validation utilities."""
+
+from typing import Dict
+
+from reana_commons.validation.parameters import build_parameters_validator
+from reana_commons.validation.compute_backends import build_compute_backends_validator
+from reana_commons.validation.utils import validate_workspace
+from reana_commons.config import AVAILABLE_WORKSPACE_PATHS
+
+from reana_server.config import SUPPORTED_COMPUTE_BACKENDS
+
+
+def validate_parameters(reana_yaml: Dict) -> None:
+    """Validate the presence of input parameters in workflow step commands and viceversa.
+
+    :param reana_yaml: REANA YAML specification.
+
+    :raises REANAValidationError: Given there are parameter validation errors in REANA spec file.
+    """
+    validator = build_parameters_validator(reana_yaml)
+    validator.validate_parameters()
+
+
+def validate_workspace_path(reana_yaml: Dict) -> None:
+    """Validate workspace in REANA specification file.
+
+    :param reana_yaml: REANA YAML specification.
+
+    :raises REANAValidationError: Given workspace in REANA spec file does not validate against
+        allowed workspaces.
+    """
+    root_path = reana_yaml.get("workspace", {}).get("root_path")
```

Comment:
This line fails for me with:

```
2022-03-04 09:35:53,615 | root | uWSGIWorker1Core1 | ERROR | Traceback (most recent call last):
  File "/code/./reana_server/rest/workflows.py", line 411, in create_workflow
    validate_workspace_path(workspace_root_path)
  File "/code/./reana_server/validation.py", line 40, in validate_workspace_path
    root_path = reana_yaml.get("workspace", {}).get("root_path")
AttributeError: 'NoneType' object has no attribute 'get'
```

If I simply pass here, then everything works in the integration tests (i.e. all the demo examples).

So a quick fix would be to do `if reana_yaml:` check here. Can you please have a look? I guess that `validate_workspace_path` might be called in some wrong order before `reana_yaml` is properly initialised?